cmd/juju/application/refresher/refresher.go

```diff
@@ -268,12 +260,9 @@ func (r baseRefresher) ResolveCharm() (*charm.URL, commoncharm.Origin, error) {
 	if err != nil {
 		return nil, commoncharm.Origin{}, errors.Trace(err)
 	}
-
-	var deployedSeries string
-	if !r.deployedBase.Channel.Empty() {
-		if deployedSeries, err = corebase.GetSeriesFromBase(r.deployedBase); err != nil {
-			return nil, commoncharm.Origin{}, errors.Trace(err)
-		}
+	deployedSeries, err := corebase.GetSeriesFromChannel(r.charmOrigin.Platform.OS, r.charmOrigin.Platform.Channel)
```

Comment:
This will be fun to merge into 3.X with base instead of series - but conflict will be easy enough to fix.